src/plugins/kibana_react/public/table_list_view/table_list_view.tsx

```diff
@@ -84,6 +85,10 @@ export interface TableListViewState {
 // This component does not try to tackle these problems (yet) and is just feature matching the legacy component
 // TODO support server side sorting/paging once title and description are sortable on the server.
 class TableListView extends React.Component<TableListViewProps, TableListViewState> {
+  static defaultProps = {
+    noItemsShowEmptyList: false,
```

Comment:
Why did you have to introduce this prop?
I find it a little confusing.
Maybe we could find a better solution?

Reply:
When there are no saved maps, we really just want to show an empty list and a button to create a new map. The default behavior resulted in this: https://github.com/elastic/kibana/issues/94638. `TableListView` does provide the option of showing a custom sub-component when there are no results, but even that's overkill for what we need. We really just want the normal component with no results. I'm open to suggestions if you can think of a better approach!

Reply:
I do have an alternative suggestion:

1. Delete `noItemsShowEmptyList`
2. Have your `noItemsFragment` set to `noItemsFragment={showEmptyList ? undefined : <Redirect to="/map" />}`
3. Inside `table_list_view` delete `renderNoItemsMessage` altogether
4. Change the condition in `renderListingOrEmptyState` to be 
```
if (this.props.noItemsFragment && !this.state.fetchError && this.hasNoItems()) {
      return this.props.noItemsFragment;
    }
```

I checked all other consumers of this component, and they all use a custom component, so this route of the code is unused.

Reply:
Sorry for the delay. Great suggestion! Works perfectly